eth/tracers/native/erc7562.go

```diff
@@ -0,0 +1,550 @@
+		f.To = nil
+	}
+	if !errors.Is(err, vm.ErrExecutionReverted) || len(output) == 0 {
+		return
+	}
+	f.Output = output
+	if len(output) < 4 {
+		return
+	}
+	if unpacked, err := abi.UnpackRevert(output); err == nil {
+		f.RevertReason = unpacked
+	}
+}
+
+type callFrameWithOpcodesMarshaling struct {
+	TypeString string `json:"type"`
+	Gas        hexutil.Uint64
+	GasUsed    hexutil.Uint64
+	Value      *hexutil.Big
+	Input      hexutil.Bytes
+	Output     hexutil.Bytes
+}
+
+type accessedSlots struct {
+	Reads           map[string][]string `json:"reads"`
+	Writes          map[string]uint64   `json:"writes"`
+	TransientReads  map[string]uint64   `json:"transientReads"`
+	TransientWrites map[string]uint64   `json:"transientWrites"`
+}
+
+type opcodeWithPartialStack struct {
+	Opcode        vm.OpCode
+	StackTopItems []uint256.Int
+}
+
+type erc7562Tracer struct {
+	config    erc7562TracerConfig
+	gasLimit  uint64
+	depth     int
+	interrupt atomic.Bool // Atomic flag to signal execution interruption
+	reason    error       // Textual reason for the interruption
+	env       *tracing.VMContext
+
+	ignoredOpcodes       map[vm.OpCode]struct{}
+	callstackWithOpcodes []callFrameWithOpcodes
+	lastOpWithStack      *opcodeWithPartialStack
+	Keccak               map[string]struct{} `json:"keccak"`
+}
+
+// catchPanic handles panic recovery and logs the panic and stack trace.
+func catchPanic() {
+	if r := recover(); r != nil {
+		// Retrieve the function name
+		pc, _, _, _ := runtime.Caller(1)
+		funcName := runtime.FuncForPC(pc).Name()
+
+		// Log the panic and function name
+		log.Error("Panic in", funcName, r)
+		debug.PrintStack()
+	}
+}
+
+// newErc7562Tracer returns a native go tracer which tracks
+// call frames of a tx, and implements vm.EVMLogger.
+func newErc7562Tracer(ctx *tracers.Context, cfg json.RawMessage, chainConfig *params.ChainConfig) (*tracers.Tracer, error) {
+	t, err := newErc7562TracerObject(ctx, cfg)
+	if err != nil {
+		return nil, err
+	}
+	return &tracers.Tracer{
+		Hooks: &tracing.Hooks{
+			OnTxStart: t.OnTxStart,
+			OnOpcode:  t.OnOpcode,
+			OnTxEnd:   t.OnTxEnd,
+			OnEnter:   t.OnEnter,
+			OnExit:    t.OnExit,
+			OnLog:     t.OnLog,
+		},
+		GetResult: t.GetResult,
+		Stop:      t.Stop,
+	}, nil
+}
+
+type erc7562TracerConfig struct {
+	StackTopItemsSize int                    `json:"stackTopItemsSize"`
+	IgnoredOpcodes    map[vm.OpCode]struct{} `json:"ignoredOpcodes"` // Opcodes to ignore during OnOpcode hook execution
+	WithLog           bool                   `json:"withLog"`        // If true, erc7562 tracer will collect event logs
+}
+
+func getFullConfiguration(partial erc7562TracerConfig) erc7562TracerConfig {
+	config := partial
+
+	if config.IgnoredOpcodes == nil {
+		config.IgnoredOpcodes = defaultIgnoredOpcodes()
+	}
+	if config.StackTopItemsSize == 0 {
+		config.StackTopItemsSize = 3
+	}
+
+	return config
+}
+
+func newErc7562TracerObject(ctx *tracers.Context, cfg json.RawMessage) (*erc7562Tracer, error) {
```

Comment:
```suggestion
func newErc7562TracerObject(cfg json.RawMessage) (*erc7562Tracer, error) {
```
ctx is unused here, no need to pass it then

Reply:
`tracers.DefaultDirectory.Register("erc7562Tracer", newErc7562Tracer, false)` expects a certain signature for the tracer that accepts that argument. That's also the case in the original callTracer